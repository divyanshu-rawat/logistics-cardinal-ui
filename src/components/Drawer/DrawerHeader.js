import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import DrawerHeaderStyled from './DrawerHeaderStyled';
import DrawerCloseButton from './DrawerCloseButton';

export const DrawerHeaderElement = styled('div')`
  flex: 1;
`;

export const DrawerHeaderButtonStyled = styled('div')`
  width: 50px;
  text-align: center;
`;

const DrawerHeader = ({ header, onClose }) => (
  <DrawerHeaderStyled>
    <DrawerHeaderElement>{header}</DrawerHeaderElement>
    <DrawerHeaderButtonStyled>
      <DrawerCloseButton onClose={onClose} />
    </DrawerHeaderButtonStyled>
  </DrawerHeaderStyled>
);

DrawerHeader.propTypes = {
  onClose: PropTypes.func.isRequired,
  header: PropTypes.node.isRequired,
};

export default DrawerHeader;
