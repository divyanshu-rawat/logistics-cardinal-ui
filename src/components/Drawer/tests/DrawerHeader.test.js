import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DrawerHeader from '../DrawerHeader';

const renderedComponent = () =>
  shallow(<DrawerHeader header={<p>Test</p>} onClose={jest.fn()} />);

describe('<DrawerHeader />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });
});
