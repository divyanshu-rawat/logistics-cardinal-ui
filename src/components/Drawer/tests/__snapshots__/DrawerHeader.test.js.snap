// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerHeader /> should render the component 1`] = `
<DrawerHeaderStyled>
  <DrawerHeader__DrawerHeaderElement>
    <p>
      Test
    </p>
  </DrawerHeader__DrawerHeaderElement>
  <DrawerHeader__DrawerHeaderButtonStyled>
    <DrawerCloseButton
      onClose={[Function]}
    />
  </DrawerHeader__DrawerHeaderButtonStyled>
</DrawerHeaderStyled>
`;
